kubernetes/template-generator/template_generator.py

```diff
@@ -0,0 +1,359 @@
+                },
+                "annotations": {
+                }
+            },
+            "spec": {
+                "containers": [
+                    {
+                        "name": "{}".format(self.container_name),
+                        "image": self.system.base_image(),
+                        "volumeMounts": []
+                    }
+                ],
+                "volumes": []
+            }
+        }
+
+        # Generate port list
+        if self.exposed_ports is not None and len(self.exposed_ports) > 0:
+            ports_template = {
+                "ports": []
+            }
+
+            for pmap in self.exposed_ports:
+                port = {
+                    "containerPort": pmap[1],
+                }
+                ports_template["ports"].append(port)
+
+            pod_template["spec"]["containers"][0].update(ports_template)
+
+        # Systemd additions
+        if self.system.is_systemd():
+            systemd_mounts = [{
+                "name": "{}-cgroup-vol".format(self.container_name),
+                "mountPath": "/sys/fs/cgroup",
+                "readOnly": True
+            }]
+
+            systemd_volumes = [{
+                "name": "{}-cgroup-vol".format(self.container_name),
+                "hostPath": {
+                    "path": "/sys/fs/cgroup"
+                }
+            }]
+
+            for mount in ["run", "tmp"]:
+                systemd_mounts.append({
+                    "name": "{}-{}-vol".format(self.container_name, mount),
+                    "mountPath": "/{}".format(mount)
+                })
+
+                systemd_volumes.append({
+                    "name": "{}-{}-vol".format(self.container_name, mount),
+                    "emptyDir": {
+                        "medium": "Memory"
+                    }
+                })
+
+            pod_template["spec"]["containers"][0]["volumeMounts"].extend(systemd_mounts)
+            pod_template["spec"]["volumes"].extend(systemd_volumes)
+
+        if yaml:
+            return self._dump_yaml(pod_template)
+
+        return pod_template
+
+    @staticmethod
+    def sanitize_container_name(name):
+        """
+            Convert container name to RFC1123 form
+
+            @param string name      a name to be sanitized
+            @return string          a sanitized name
+
+            @throw SanitizeException
+        """
+        if not len(name):
+            raise SanitizeException("Name cannot be empty")
+
+        # RE: [a-z0-9]([-a-z0-9]*[a-z0-9])? / RFC1123
+
+        # Replace invalid characters and remove leading -
+        found_name = re.findall("^-*([a-z0-9][a-z0-9-]*)*$", re.sub("[^0-9a-z]", "-", name))
+
+        if len(found_name) >= 1:
+            rev_name = found_name[0][::-1]
+            return re.findall("^-*([a-z0-9][a-z0-9-]*)*$", rev_name)[0][::-1]
+        else:
+            raise SanitizeException("Name could not be sanitized")
```

Comment:
This isn't helpful error message at all, how am I, as a user, supposed to know how to fix this error?